### PR TITLE
[GLT-3911] Renamed "short name" and "alias" fields in Project.

### PR DESCRIPTION
**Pull request by aryan2gandevia**
Jira ticket or GitHub issue: https://jira.oicr.on.ca/browse/GLT-3911

✅ Includes a change file

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/view/ListLibraryAliquotView.java [hunk 1]
```diff
@@ -231,11 +231,11 @@ public void setAliquotDesignCode(LibraryDesignCode aliquotDesignCode) {
   }
 
   public String getProjectShortName() {
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/view/ListLibraryAliquotView.java:233]: Should this change to `getProjectCode()` ?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/view/ListLibraryAliquotView.java [hunk 2]
```diff
@@ -231,11 +231,11 @@ public void setAliquotDesignCode(LibraryDesignCode aliquotDesignCode) {
...
-    return getProjectAttribute(ParentProject::getShortName);
+    return getProjectAttribute(ParentProject::getCode);
   }
 
   public String getProjectAlias() {
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/view/ListLibraryAliquotView.java:237]: Should this change to `getProjectTitle()`?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/view/transfer/ListTransferViewPool.java [hunk 3]
```diff
@@ -26,7 +26,7 @@ public class ListTransferViewPool extends ListTransferViewItem {
     private static final String SHORT_NAME = "Undetermined (pool)";
 
     @Override
-    public String getShortName() {
+    public String getCode() {
       return SHORT_NAME;
```
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/view/transfer/ListTransferViewPool.java:30]: Do you want to modify this to `CODE` for consistency?

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java [hunk 4]
```diff
@@ -517,8 +517,8 @@ private static SampleDto copySampleFields(@Nonnull Sample from, @Nonnull SampleD
     dto.setAlias(from.getAlias());
     dto.setProjectId(from.getProject().getId());
     dto.setProjectName(from.getProject().getName());
-    dto.setProjectAlias(from.getProject().getAlias());
-    dto.setProjectShortName(from.getProject().getShortName());
+    dto.setProjectAlias(from.getProject().getTitle());
+    dto.setProjectShortName(from.getProject().getCode());
```
Comment (callunity) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java:521]: Please modify `SampleDto` to change `get/setProjectAlias` -> `get/setProjectTitle` and `get/setProjectShortName` -> `get/setProjectCode`

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java [hunk 5]
```diff
@@ -1403,7 +1403,7 @@ public static LibraryDto asDto(@Nonnull Library from, boolean includeBoxPosition
     dto.setParentSampleAlias(from.getSample().getAlias());
     dto.setProjectId(from.getSample().getProject().getId());
     setString(dto::setProjectName, from.getSample().getProject().getName());
-    setString(dto::setProjectShortName, from.getSample().getProject().getShortName());
```
Comment (callunity) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java:1406]: Please modify `LibraryDto` to change `get/setProjectShortName` -> `get/setProjectCode`

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java [hunk 6]
```diff
@@ -1715,7 +1715,7 @@ public static LibraryAliquotDto asDto(@Nonnull LibraryAliquot from, boolean incl
         if (sample.getProject() != null) {
           setLong(dto::setProjectId, project.getId(), true);
           setString(dto::setProjectName, project.getName());
-          setString(dto::setProjectShortName, project.getShortName());
+          setString(dto::setProjectShortName, project.getCode());
```
Comment (callunity) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java:1718]: Please modify `LibraryAliquotDto` to change `get/setProjectShortName` -> `get/setProjectCode`

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/context/ExternalUriBuilder.java [hunk 7]
```diff
@@ -41,7 +44,7 @@ public Map<String, String> getUris(Run run) {
   private String expandProjectUrl(String uriWithPlaceholders, Project project) {
     return uriWithPlaceholders.replaceAll(ID_PLACEHOLDER, String.valueOf(project.getId()))
         .replaceAll(NAME_PLACEHOLDER, project.getName())
-        .replaceAll(SHORTNAME_PLACEHOLDER, project.getShortName());
+        .replaceAll(SHORTNAME_PLACEHOLDER, project.getCode());
```
Comment (callunity) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/context/ExternalUriBuilder.java:47]: `SHORTNAME_PLACEHOLDER` -> `CODE_PLACEHOLDER` for consistency?

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/context/MisoAppListener.java [hunk 8]
```diff
@@ -173,39 +169,46 @@ private NamingScheme getConfiguredNamingScheme(String namespace, Map<String, Str
...
+        scheme::setNameValidator);
     applyValidator(namespace + ".validator.sample.alias", misoProperties, resolver::getSampleAliasValidator,
         scheme::setSampleAliasValidator);
     applyValidator(namespace + ".validator.library.alias", misoProperties, resolver::getLibraryAliasValidator,
         scheme::setLibraryAliasValidator);
-    applyValidator(namespace + ".validator.libraryaliquot.alias", misoProperties, resolver::getLibraryAliquotAliasValidator,
+    applyValidator(namespace + ".validator.libraryaliquot.alias", misoProperties,
+        resolver::getLibraryAliquotAliasValidator,
         scheme::setLibraryAliquotAliasValidator);
     applyValidator(namespace + ".validator.project.shortName", misoProperties, resolver::getProjectShortNameValidator,
```
Comment (callunity) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/context/MisoAppListener.java:195]: Should probably change `".validator.project.shortName"` to `".validator.project.code"` for consistency, and I believe `resolver::getProjectShortNamevalidator` should be `resolver::getProjectCodeValidator`

File: miso-web/src/main/webapp/scripts/form_project.js [hunk 9]
```diff
@@ -85,12 +85,12 @@ FormTarget.project = (function ($) {
               },
             },
             {
-              title: "Short Name",
-              data: "shortName",
+              title: "Code",
+              data: "doe3",
```
Comment (callunity) [inline miso-web/src/main/webapp/scripts/form_project.js:89]: `doe3` -> `code`

**Review by callunity — COMMENTED** [review]
Nice work on a monster refactor! I've added a couple of requests, mostly for consistency. 

Please also add in a `changes/note_GLT-3911.md` file with the following:

If your `miso.properties` file contains the property `miso.naming.validator.project.shortName` or `miso.naming.validator.project.shortName.duplicates`, these will need to be changed:
| existing value | new value |
|--------------------|---------------|
| `miso.naming.validator.project.shortName` | `miso.naming.validator.project.code` |
| `miso.naming.validator.project.shortName.duplicates` | `miso.naming.validator.project.code.duplicates` |


`pinery-miso`'s `queryAllSamples.sql` also has a few stray `shortName` that will need to be updated, which should help with a number of the failing tests.

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/view/EditProjectController.java [hunk 10]
```diff
@@ -76,6 +76,15 @@ public ModelAndView setupForm(ModelMap model) throws IOException {
     return setupForm(new ProjectImpl(), model);
   }
 
+  @GetMapping("/shortname/{shortName}")
+  public ModelAndView byProjectShortName(@PathVariable String shortName, ModelMap model) throws IOException {
+    Project project = projectService.getProjectByCode(shortName);
+    if (project == null) {
+      throw new NotFoundException("No project found for short name " + shortName);
+    }
+    return new ModelAndView("forward:/project/code/%s".formatted(shortName), model);
```
Comment (djcooke) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/view/EditProjectController.java:85]: This method should only do the redirect. Validation will be handled after the redirect in `byProjectCode` below

Comment (djcooke) [reply miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/view/EditProjectController.java:85]: Also, it should be "redirect:", not "forward:", sorry!

File: changes/note_#2_GLT-3911.md [hunk 11]
```diff
@@ -0,0 +1 @@
+Links for external projects that use their short names changed from (`http://example.com/project/{shortName}`) to using their code (`http://example.com/project/{code}`)
```
Comment (djcooke) [inline changes/note_#2_GLT-3911.md:1]: This is not explaining what needs to be done like your other note is. Something like "If your `miso.properties` file specifies `miso.project.report.links`, any `{shortName}` placeholders in this must be updated to `{code}`"

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/view/EditProjectController.java [hunk 12]
```diff
@@ -82,7 +82,7 @@ public ModelAndView byProjectShortName(@PathVariable String shortName, ModelMap
     if (project == null) {
       throw new NotFoundException("No project found for short name " + shortName);
     }
-    return new ModelAndView("forward:/project/code/%s".formatted(shortName), model);
+    return new ModelAndView("redirect:/project/code/%s".formatted(shortName), model);
```
Comment (djcooke) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/view/EditProjectController.java:85]: Everything above this line should be removed from the method (validation already happens after redirect)

**Review by djcooke — APPROVED** [review]

**Pull request merged**